src/main/java/ch/njol/skript/doc/JSONGenerator.java

```diff
@@ -383,40 +460,70 @@ private static JsonArray cleanPatterns(String... strings) {
 	}
 
 	/**
-	 * Writes the documentation JsonObject to an output path
+	 * Gets the json object representing the addon.
 	 *
-	 * @param outputPath the path to write the documentation to
-	 * @param jsonDocs   the documentation JsonObject
+	 * @return The json object representing the addon.
 	 */
-	private void saveDocs(Path outputPath, JsonObject jsonDocs) {
-		try {
-			Files.writeString(outputPath, GSON.toJson(jsonDocs));
-		} catch (IOException exception) {
-			//noinspection ThrowableNotThrown
-			Skript.exception(exception, "An error occurred while trying to generate JSON documentation");
+	private JsonObject getSource() {
+		JsonObject object = new JsonObject();
+
+		object.addProperty("name", source.name());
+
+		Plugin plugin = Bukkit.getServer().getPluginManager().getPlugin(source.name());
+		if (plugin != null) {
+			object.addProperty("version", plugin.getDescription().getVersion());
+		} else {
+			try {
+				plugin = JavaPlugin.getProvidingPlugin(source.source());
+				object.addProperty("version", plugin.getDescription().getVersion());
+			} catch (Exception ex) {
+				object.add("version", null);
+			}
 		}
```

Comment:
```suggestion
		Plugin plugin = Bukkit.getServer().getPluginManager().getPlugin(source.name());
		if (plugin == null) {
			try {
				plugin = JavaPlugin.getProvidingPlugin(source.source());
			} catch (Exception ignored) { }
		}
		object.addProperty("version", plugin == null ? null : plugin.getDescription().getVersion());
```
This should work I think. Might be better to catch a more specific exception if possible.